### ISSUE TITLE
Block inserter toggle: avoid dynamic label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68536]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/pull/68531 

When the `Show button text labels` is enabled, the label of the block inserter toggle in the top bar is shortened from `Block Inserter` to `Add`.
However, when opening the block inserter, the label dynamically changes from `Add` to `Close`. Screenshots:

![Image](https://github.com/user-attachments/assets/564e1aa0-7e7a-4993-8fb9-897407ced2a0)

While I do understand the intent to provide a _visual_ indication, control labels should not be used for mere visual purposes. Their purpose is to unambiguously communicate _what_ a control does or is about and should not change dynamically. 

The pressed state of the button is already communicated by a color inversion and an appropriate `aria-pressed` attribute set on the button (although the usage of `aria-pressed` is slightly incorrect in this case). The button also uses an `aria-expanded` attribute. Semantically, there's no need to add anything else and the label should not change. I'd argue that also visually the label should not change.

Additionally, when `Show button text labels` is disabled, the label of the button is `Block Inserter` which is way too technical for users. Should a good UI communicate obscure technical terminology or rather provide users with a simple, understandable, label? I'd vote for the latter and use `Add` also when the preference is disabled.

Note:
in https://github.com/WordPress/gutenberg/pull/68531  I'm implementing this 'dynamic' label in the Widgets editor only to not introduce a new inconsistency across the editors. If https://github.com/WordPress/gutenberg/pull/68531  gets merged, this issue should be addressed also in the Widgets editor.

### Step-by-step reproduction instructions

- Go to the post or site editor.
- Hover of focus the block inserter button.
- Observe the tooltip text, which is also the label of the button, is `Block Inserter`.
- Enable the preference `Show button text labels` from the top bar Options > Preferences > Accessibility.
- Observe the button text is now `Add`.
- Open the inserter.
- - Observe the button text is now `Close`.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Mamaduka** [comment]
I think that the label "Add" lacks context. What are we adding?

@afercia, why do you think "Block Inserter" is too technical for users? As much as I would like labels to be 100% intuitive and non-technical, I don't think it can be entirely avoided. You'll have to learn a few new terms with every software program.

**Comment by afercia** [comment]
@mamaduka well, first of all in the main inserter is now possible to add Blocks, Patterns, and Media.
As such, 'Block inserter' feels obsolete and not accurate because it's not just about 'blocks'. 

'Inserter' is way too technical for my taste and also difficult to translate. For example, in Italian it is translated to:

`Pannello di inserimento dei blocchi`

![Image](https://github.com/user-attachments/assets/9b659beb-0509-4f1c-990f-fda8b43ed187)

Literally: `Panel of insertion of the blocks`, which is too long and noisy. It sill refers to 'blocks'. We can't blame translators for such a non-ideal translation. The original string isn't ideal in the first place. I'd be curious to check how it is translated in other languages. I guess it's non-ideal for many other languages as well, which would prove ths original name has a problem in the first place.

If `Add` isn't ideal, then why we would want to use it only for the case when `Show button text labels` is enabled? If there's agreement that it's not ideal, it should never be used. 

The button icon is a plus. It hints to the concept of 'adding', so tha tI would think that's the concept that should be communicated: 'Add stuff', whether it's a block, pattern, media, or anything else.

Note: some previous considerations to take into account were also reported at https://github.com/WordPress/gutenberg/issues/61483

**Comment by Mamaduka** [comment]
> first of all in the main inserter is now possible to add Blocks, Patterns, and Media.
As such, 'Block inserter' feels obsolete and not accurate because it's not just about 'blocks'.

Everything is a block at the end. Patterns and Media are just "sub-categories"; they are inserted as block(s) at the end.

> I'd be curious to check how it is translated in other languages.

The old "toggle" version has the following translation in Georgian - `ბლოკის ჩასმის ჩართვა`. I'm pretty sure the localization team will use `ბლოკის ჩასმა` for the current label, which translates into "Add block". Georgian doesn't have a lot of modern words and has to get by what we have 😄

> If Add isn't ideal, then why we would want to use it only for the case when Show button text labels is enabled? If there's agreement that it's not ideal, it should never be used

I didn't say that the current text version is ideal. Let's find a version of a label that works for both cases.

**Comment by afercia** [comment]
I'm all for finding a new name of the feature that is simpler and more understandable :) 
My point about the translations is that they prove the English name isn't ideal in the first place.
